app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -26,7 +26,13 @@ def create
     private
 
     def confirm_two_factor_enabled
-      return if confirmation_context? || current_user.phone_enabled?
+      phone_enabled = current_user.phone_enabled?
+
+      return if confirmation_context? || phone_enabled
+
+      if current_user.two_factor_enabled? && !phone_enabled? && user_signed_in?
+        return redirect_to user_two_factor_authentication_url
+      end
```

Comment:
Good catch! Can we write these tests please?

- totp_enabled but not phone enabled should not be able to access `phone_setup` page directly after signing in with just email and password
- piv_cac_enabled but not phone enabled should not be able to access `phone_setup` page directly after signing in with just email and password
- totp_enabled but not phone enabled should not be able to access `phone_setup` page indirectly by visiting `login/two_factor/sms` after signing in with just email and password
- piv_cac_enabled but not phone enabled should not be able to access `phone_setup` page indirectly by visiting `login/two_factor/sms` after signing in with just email and password
- totp_enabled but not phone enabled should not be able to access `phone_setup` page indirectly by visiting `login/two_factor/voice` after signing in with just email and password
- piv_cac_enabled but not phone enabled should not be able to access `phone_setup` page indirectly by visiting `login/two_factor/voice` after signing in with just email and password